yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/block/FileOptions.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * FileOptions.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.relational.yamltests.block;
+
+import com.apple.foundationdb.relational.yamltests.Matchers;
+import com.apple.foundationdb.relational.yamltests.YamlExecutionContext;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Assumptions;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Block that configures aspects of the test that do not require a connection yet.
+ * <p>
+ *     This supports the following sub-commands:
+ *     <ul>
+ *         <li>{@code supported_version}: if this is set, it will disable the test when running against a server
+ *         older than a specific version. The special yaml tag {@code !current_version} can be used to indicate that
+ *         it should work against the current code, but is not expected to work with any older versions.
+ *         In the future, the release script will update these automatically to the version being released.
+ *         <p>
+ *             <b>Example:</b>
+ *             <pre>{@code
+ *                 ---
+ *                 options:
+ *                     supported_version: !current_version
+ *             }</pre>
+ *     </ul>
+ */
+public class FileOptions {
```

Comment:
Not that it is necessary, do you think there is an advantage in ensuring that the FileOptions is strictly the first block in a YAMSQL file (if one is present)? However, that does not matter currently, as it would simply fail the assumption if conditions are not met at parse time but it would still be good to have such a syntax rule as a way of enforcing developers to defines options the first thing in the file (for better readability?). 

We may also want to avoid having multiple fileOption blocks in the file as a way to not allow having multiple option values.

Reply:
I created a followup issue to add those restrictions: #3045